### PR TITLE
Add random and slideshow modes to Catpacks

### LINKED ISSUE #693: Random cat selection
### Role

Because it's cats

### Suggestion

Add an option to show a random cat every time you load the launcher

### Benefit

More cats :)

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

This is reccomanded only if more cats are added or you will cycle between 2 images basically
Also i consider adding the cat selection menu by right clicking the cat button

### LINKED ISSUE #1816: (Windows) Cats are visibly "crusty" or pixellated
### Operating System

Windows

### Version of Prism Launcher

Prism Launcher 8.0

### Version of Qt

QT 6.6.0

### Description of bug

Currently cats built into prism on windows appear "crustier" and more pixellated than ones on linux.

image
image

### Steps to reproduce

- Have prism launcher on windows (I'm personally on Windows 11 22H2)
- Press the cat button
- Crumchy

### Suspected cause

most likely relating to a scale cat fix made recently by trial, their words not mine. Was told to make a git issue about it by them

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### LINKED ISSUE #1820: Add better options/functions for CatPack customization
### Role

I (intend) to make catpacks in the future

### Suggestion

I want to have 2 new functions added to catpacks that will make producing catpacks easier and more optimized

### Benefit

Less bulky catpacks, saves time for the catpack creator

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

With the current system for catpacks, you have to specify a day and month for each image. Currently this system works for a small scale catpack with 1 image and a few holiday variants, but might become problematic once you try to make a catpack with multiple images that you want to change every day.

The only method at the moment is to specify every day you want that particular image to show, and if you have 4 images in a catpack that you want to have as a sort of slideshow, then you have to do this for every single image, in order.

I have a small suggestion on how this can be improved with 2 new options/functions

1.  "Random", Catpack has a folder in it called either random or just "default_image" and the json file has the ability to select a random image from that file, either every relaunch of prism or every day (gonna leave that up to you)

2. "Slideshow" Catpack has a folder called slideshow, where the images are numbered (like 1.png, 2.png etc) and the file goes through the set of images in order. When it reaches the end, it starts over.

Unsure how possible this is, but it would save me and others alot of time when we are creating catpacks, and should also reduce the json file's size since we don't have to have the same lines repeated over and over again.

I'm not asking you to completely remove the current system, since its good for holiday specific images but I think it should be kept just for specifying holiday images and small scale catpacks with one default image.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #693
fixes #1820 
and hopefully fixes #1816 (needs testing)

So here are the additions to the catpacks:
- if the default path points to a directory then we will display one cat per day from that directory in a slideshow manner (the files are sorted alphabetically then the file index is chosen by the following formula `theDayOfTheYear % numFiles`)
- if that directory is named `random` (case insensitive) then a new cat is selected on each prism restart or cat button toggle 
- make sure we apply smooth transformation when scaling of the cat

### CODE REVIEW

**Comment by SympathyTea** [comment]
super excited about this one, hopefully it works out

File: launcher/ui/themes/CatPack.cpp [hunk 1]
```diff
@@ -117,5 +120,21 @@ QString JsonCatPack::path(QDate now)
...
+
+    QStringList supportedImageFormats;
+    for (auto format : QImageReader::supportedImageFormats()) {
+        supportedImageFormats.append("*." + format);
+    }
+
+    auto files = QDir(m_default_path).entryInfoList(supportedImageFormats, QDir::Files, QDir::Name);
+    if (files.length() == 0)
+        return "";
+    auto idx = now.dayOfYear() % files.length();
```
Comment (TheKodeToad) [inline launcher/ui/themes/CatPack.cpp]:
```suggestion
    auto idx = (now.dayOfYear() - 1) % files.length();
```
First day = 1, first file = 0, that doesn't make sense and the you will always see the cat of tommorow

**Review by TheKodeToad — CHANGES_REQUESTED** [review]
Looks pretty good!! Just one thing I noticed.

**Comment by TheKodeToad** [comment]
Isn't choosing based on alphabetical order a little weird? It probably suffers from the classic issue of sorting numbers like e.g. `1`, `10`, `11`, `12`, `2`, `3`, `4`, `5` and requiring more concrete names would be better imo

**Comment by Trial97** [comment]
Well yeah if people want more control over it they can add a 0 before the digit like:
`01`, `02`, `03`, ...,`10` 
If they have more that 100 cats then add more zeros

**Review by TheKodeToad — APPROVED** [review]

**Review by getchoo — APPROVED** [review]

**Pull request merged**